frontend/lib/Repository/UserRepository/UserGateway.dart

```diff
@@ -24,15 +22,14 @@ class UserGateway {
   }
 
   Future<AuthResponse> authenticateUser(AuthenticateForm form) async {
-    final String? access = await secureStorage.read(key: accessKey);
-    final String? refresh = await secureStorage.read( key: refreshKey);
+    String? access = await secureStorage.read(key: accessKey);
+    String? refresh = await secureStorage.read(key: refreshKey);
 
     if (access != null && refresh != null) {
       //print("FROM MEMORY"); // for reviewer
       user =
           User(AuthTokenResponse(accessToken: access, refreshToken: refresh));
-      return AuthResponse(
-          success: true, tokens: user.tokenResponse);
+      return AuthResponse(success: true, tokens: user.tokenResponse);
```

Comment:
Ta funkcja powinna wyglądać tak. Jeżeli jesteśmy na ekranie logowania/rejestrowania to ZAWSZE chcemy zapisać nowy token. Jeżeli tego nie naprawimy to apka nie będzie działać :crying_cat_face: 
```dart
Future<AuthResponse> authenticateUser(AuthenticateForm form) async {
      final AuthResponse response = await authGetaway.authenticateUser(form);
      user = response.success ? User(response.tokens!) : User.empty();
      await secureStorage.write(
          key: accessKey, value: response.tokens!.accessToken);
      await secureStorage.write(
          key: refreshKey, value: response.tokens!.refreshToken);
      return response;
}
```